### PR TITLE
Rdm 6321 start case creation as citizen-FTA

### PR DESCRIPTION
**Pull request by shrivsksavit**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-6321

### Change description ###
Implement Functional Tests for 'Start case creation as Citizen' operation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/aat/resources/features/F-058/F-058.feature [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  Scenario: must return 401 when request provides invalid authentication credentials
+    Given   a user with [an active profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [contains a dummy user id]
+    And     it is submitted to call the [Start case creation as Citizen] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-401]
+    And     the response has all the details as expected
+
+
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-058/F-058.feature]: pls remove extra empty line

File: src/aat/resources/features/F-058/F-058.feature [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  Scenario: must return 403 when the request contains a dummy user id
+    Given   a user with [an active profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [contains a dummy user id]
+    And     it is submitted to call the [Start case creation as Citizen] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-403]
+    And     the response has all the details as expected
+
+
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-058/F-058.feature]: pls remove extra empty line

File: src/aat/resources/features/F-058/F-058.feature [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  @S-240 @Ignore # Response code mismatch, expected: 400, actual: 500
+  Scenario: must return 400 when request contains a malformed Case Type ID
+    Given   a user with [an active profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [contains a malformed Case Type ID]
+    And     it is submitted to call the [Start case creation as Citizen] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-400]
+    And     the response has all the details as expected
+
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-058/F-058.feature]: pls remove extra empty line

File: src/aat/resources/features/F-058/F-058.feature [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  Scenario: must return 404 when request contains a non-existing jurisdiction ID
+    Given   a user with [an active profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [contains a non-existing jurisdiction ID]
+    And     it is submitted to call the [Start case creation as Citizen] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-404]
+    And     the response has all the details as expected
+
+
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-058/F-058.feature]: pls remove extra empty line

File: src/aat/resources/features/F-058/F-058.feature [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  @S-519
+  Scenario: must return 404 when request contains a non-existing Event ID
+    Given   a user with [an active profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [contains a non-existing Event ID]
+    And     it is submitted to call the [Start case creation as Citizen] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-404]
+    And     the response has all the details as expected
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-058/F-058.feature]: eol at end of file

File: src/aat/resources/features/F-058/S-239.td.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    "_extends_": "Common_403_Response",
+    "headers": {
+      "Content-Encoding": "gzip",
+      "Content-Type": "application/json;charset=UTF-8"
+    },
+    "body": {
+      "message": "Access Denied"
+    }
+  }
+}
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-058/S-239.td.json]: eol at end of all files please

File: src/aat/resources/features/F-058/F-058.feature [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    Given an appropriate test context as detailed in the test data source
+
+  @S-241
+  Scenario: must start case creation process successfully for correct inputs
+    Given   a user with [an active profile in CCD]
+    When    a request is prepared with appropriate values
+    And     it is submitted to call the [Start case creation as Citizen] operation of [CCD Data Store]
+    Then    a positive response is received
+    And     the response [code is HTTP-200]
+    And     the response has all the details as expected
```
Comment (aashishaggarwal123) [inline src/aat/resources/features/F-058/F-058.feature]: Please use ->> 'the response has all other details as expected'

File: src/aat/resources/features/F-058/F-058.feature [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    And     it is submitted to call the [Start case creation as Citizen] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-403]
+    And     the response has all the details as expected
+
+  @S-238 @Ignore # Response code mismatch, expected: 403, actual: 200
+  Scenario: must return 403 when the request contains a jurisdiction id user is unauthorised to access
+    Given   a user with [an active profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [contains a jurisdiction id user is unauthorised to access]
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-058/F-058.feature]: contains a jurisdiction id user is unauthorised to access. It should be free flow

File: src/aat/resources/features/F-058/F-058.feature [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    And     it is submitted to call the [Start case creation as Citizen] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-403]
+    And     the response has all the details as expected
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-058/F-058.feature]: should use following DSL: 
the response has all other details as expected
--

File: src/aat/resources/features/F-058/F-058.feature [hunk 10]
```diff
@@ -0,0 +1,94 @@
+@F-058
+Feature: Start case creation as Citizen
```
Comment (aashishaggarwal123) [inline src/aat/resources/features/F-058/F-058.feature]: "Feature: F-058 : Start case creation as Citizen"

File: src/aat/resources/features/F-058/S-517.td.json [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+  "title": "must return 404 when request contains a non-existing jurisdiction ID",
+
+  "specs": [
+    "an active profile in CCD",
+    "contains a non-existing jurisdiction ID",
+    "code is HTTP-403"
+  ],
+  "request": {
+    "pathVariables": {
+      "jid": "Dummy"
```
Comment (aashishaggarwal123) [inline src/aat/resources/features/F-058/S-517.td.json]: please use arbitrary value which won't clash. may be something like 'eyJhbGciOiJIUzUxMiJ9'

File: src/aat/resources/features/F-058/F-058.feature (hunk 8 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-058/F-058.feature]: sorry this one already approved by Mutlu in RDM-6320 as per his suggestion

**Pull request merged**